src/Compilers/Core/Portable/Emit/EditAndContinue/SynthesizedTypeMaps.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Emit;
+
+internal readonly struct SynthesizedTypeMaps(
+    ImmutableDictionary<AnonymousTypeKey, AnonymousTypeValue>? anonymousTypeMap,
+    ImmutableDictionary<SynthesizedDelegateKey, SynthesizedDelegateValue>? anonymousDelegates,
+    ImmutableDictionary<string, AnonymousTypeValue>? anonymousDelegatesWithIndexedNames)
```

Comment:
Consider using `ImmutableSegmentedDictionary` instead. It appears we're mainly using these for reads, not appends, and `ImmutableSegmentedDictionary` is `O(1)` while `ImmutableDictionary` is `O(log n)` for that operation.